src/library/importer.py

```diff
@@ -17,14 +24,54 @@ def import_quests_to(*, destination_schema, quest_import_ids):
 
     dry_run = False
     with schema_context(destination_schema):
-        res = QuestResource().import_data(export_data, dry_run=dry_run)
+        # Explicitly import the campaign as well
+        result = QuestResource().import_data(export_data, dry_run=dry_run, import_campaign=True)
 
         object_ids = []
-        for row in res.rows:
+        for row in result.rows:
             if row.new_record:
                 object_ids.append(row.object_id)
-        # Set visible_to_students to False for imported quests
-        # since the imported quests will be orphans
+        # Set visible_to_students (published) to False for imported quests,
+        # using update() for efficiency and to avoid triggering full model validation or signals,
+        # since we're only flipping a simple boolean flag on multiple objects at once.
         Quest.objects.filter(pk__in=object_ids).update(visible_to_students=False)
 
-    return res
+        category = Category.objects.filter(import_id=campaign_import_id).first()
+        if category:
+            category.active = False
+            category.full_clean()
+            category.save()
+
+    return result
+
+
+def import_quest_to(*, destination_schema, quest_import_id):
+    """
+    Imports a single quest into the destination schema without importing its campaign.
+
+    Automatically sets imported quest to not be visible to students (not published)
+
+    Args:
+        destination_schema (str): The schema to import the quest into.
+        quest_import_id (UUID): The import ID of the quest to import.
+    """
+    with library_schema_context():
+        quest = Quest.objects.get(import_id=quest_import_id)
+
+        export_data = QuestResource().export([quest])
+
+    with schema_context(destination_schema):
+        result = QuestResource().import_data(
+            export_data,
+            dry_run=False,
+            raise_errors=True,
+            use_transactions=True,
+        )
+
+        imported_quest = Quest.objects.get(import_id=quest_import_id)
+        if imported_quest:
```

Comment:
Coderabbitai is right, there's no point checking if imported quest exists after the get(), because the get() would have thrown an exception if it didn't